src/Haskell/Ide/Engine/Plugin/GhcMod.hs

```diff
@@ -290,6 +293,66 @@ cmp a b
 isSubRangeOf :: Range -> Range -> Bool
 isSubRangeOf (Range sa ea) (Range sb eb) = sb <= sa && eb >= ea
 
+
+
+data DocumentPosition =
+  DocPos {
+        docFile :: Uri
+      , docPos  :: Position
+      } deriving (Eq,Generic,Show)
+
+instance FromJSON DocumentPosition where
+  parseJSON = genericParseJSON $ defaultOptions { fieldLabelModifier = camelTo2 '_' . drop 3}
+instance ToJSON DocumentPosition where
+  toJSON = genericToJSON $ defaultOptions { fieldLabelModifier = camelTo2 '_' . drop 3}
+
+splitCaseCmd ::CommandFunc DocumentPosition WorkspaceEdit
+splitCaseCmd = CmdSync $ \(DocPos uri pos) -> do
+  splitCaseCmd' uri pos
+
+splitCaseCmd' :: Uri -> Position -> IdeGhcM (IdeResult WorkspaceEdit)
+splitCaseCmd' uri position =
+  pluginGetFile "splitCaseCmd: " uri $ \path -> do
+    origText <- GM.withMappedFile path $ liftIO . T.readFile
```

Comment:
This version of the file may differ from the version we have a TypecheckedModule for. Use `newPosToOld` and `oldPosToNew` to correlate positions across versions of the file.